pkg/ccl/backupccl/restore_planning.go

```diff
@@ -320,21 +320,30 @@ func allocateDescriptorRewrites(
 	descriptorCoverage tree.DescriptorCoverage,
 	opts tree.RestoreOptions,
 	intoDB string,
+	newDBName string,
 ) (DescRewriteMap, error) {
 	descriptorRewrites := make(DescRewriteMap)
+
+	// additional variables to handle into_db option
 	var overrideDB string
-	var renaming bool
+	var newParentDB bool
 	if opts.IntoDB != nil {
 		overrideDB = intoDB
-		renaming = true
+		newParentDB = true
+	}
+
+	if (opts.NewDBName == nil && len(newDBName) != 0) ||
```

Comment:
yeah, that makes sense. I added this check because I was getting confused with the variables associated with the IntoDB option. Specifically, why we do need to create `overridingDB` (the name you suggest in the conversation above) and `overrideDB` , only after checking that `opts.IntoDB` is non null? i.e.

```
// overrideDB holds the value of the into_db restore option,
// when restoring a table into a specified database.
	var overrideDB string
	var overridingDB bool
	if opts.IntoDB != nil {
		overrideDB = intoDB
		overridingDB = true
	}
```

If we don't need to check the opt.IntoDB value once again, then why create `OverrideDB` and its bool counterpart, when we already have `intoDB`? We can check that `intoDB` is null easily, which means these additional variables are unnecessary.